### PR TITLE
slots warning fix

### PR DESCRIPTION
**Pull request by rajeeja**
https://docs.xarray.dev/en/stable/whats-new.html\?highlight\=slots\#v0-13-0-17-sep-2019

    object.__setattr__(self, name, value)
../uxarray/core/dataset.py:42: in uxgrid
    self._uxgrid = ugrid
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[AttributeError("'UxDataArray' object attribute '_uxgrid' is read-only") raised in repr()] UxDataset object at 0x15ff32640>
name = '_uxgrid', value = <uxarray.core.grid.Grid object at 0x16c7103d0>

    def __setattr__(self, name: str, value: Any) -> None:
        """Objects with ``__slots__`` raise AttributeError if you try setting an
        undeclared attribute. This is desirable, but the error message could use some
        improvement.
        """
        try:
>           object.__setattr__(self, name, value)
E           AttributeError: 'UxDataset' object attribute '_uxgrid' is read-only

### CODE REVIEW

**Comment by rajeeja** [comment]
This PR is now ready for review.

File: uxarray/core/dataarray.py [hunk 1]
```diff
@@ -26,10 +27,9 @@ def uxgrid(self):
 
     # a setter function
     @uxgrid.setter
-    def uxgrid(self, ugrid):
-
-        self._uxgrid = ugrid
+    def uxgrid(self, grid_obj):
+        self._uxgrid = grid_obj
```
Comment (philipc2) [inline uxarray/core/dataarray.py:31]: I remember you mentioning something about our naming scheme for the grid object, especially having consistency throughout the codebase. I'm torn between either calling it `ugrid` (short for unstructured grid and matches the UGRID conventions) or `uxgrid` (short for Uxarray Grid). 

Make sure any changes to this are properly represented in both the DataArray and Dataset modules and for the setters and getters.

**Review by philipc2 — CHANGES_REQUESTED** [review]
Implementation looks good, left a few notes about some comment and style concerns.

File: uxarray/core/dataarray.py (hunk 1 above)
Comment (rajeeja) [reply uxarray/core/dataarray.py]: gotcha, changed it to `ugrid_obj`

Comment (philipc2) [reply uxarray/core/dataarray.py]: works with me

**Review by philipc2 — APPROVED** [review]

**Pull request merged**